syftlib/src/main/java/org/openmined/syft/execution/JobDownloader.kt

```diff
@@ -0,0 +1,181 @@
+package org.openmined.syft.execution
+
+import android.util.Log
+import io.reactivex.Single
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.processors.PublishProcessor
+import org.openmined.syft.domain.SyftConfiguration
+import org.openmined.syft.networking.datamodels.ClientConfig
+import org.openmined.syft.proto.SyftModel
+import org.openmined.syft.utilities.FileWriter
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicReference
+
+private const val TAG = "JobDownloader"
+
+@ExperimentalUnsignedTypes
+internal class JobDownloader {
+    fun downloadData(
+        workerId: String,
+        config: SyftConfiguration,
+        requestKey: String?,
```

Comment:
It's better to make this non nullable. Since it's an independent function call now

Reply:
Ok. I wanted to keep it the same but if it's ok to be non nullable, much better :)